### PR TITLE
Pnpm migration

### PR DESCRIPTION
**Pull request by manojVivek**

### CODE REVIEW

**Review by brancz — APPROVED** [review]
Very nice, I was a bit worried the earley parser was going to cause trouble, but looks like it was pretty smooth. Nice work!

**Review by yomete — APPROVED** [review]
nice work!

**Review by metalmatze — COMMENTED** [review]
amazing! :rocket:

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -61,13 +61,17 @@ jobs:
       with:
         go-version-file: .go-version
 
+    - uses: pnpm/action-setup@a3252b78c470c02df07e9d59298aecedc3ccdd6d # v3.0.0
+      with:
+        version: 8.15.6
```
Comment (maxbrunet) [inline .github/workflows/build-test.yml:66]: Could you set the version from the env and add a comment for Renovate like we do for other tools?

```suggestion
        version: ${{ env.PNPM_VERSION }}
```

```yaml
env:
  # renovate: datasource=npm depName=pnpm versioning=npm
  PNPM_VERSION: 8.15.6
```

**Review by maxbrunet — COMMENTED** [review]
LGTM 🚀 

just one comment

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (manojVivek) [reply .github/workflows/build-test.yml]: Thanks for catching this, updated other workflow files as well.

**Pull request merged**